lib/hooks/useOSDetection.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react";
+
+function useOSDetection(): boolean {
+  const [isMac, setIsMac] = useState<boolean>(false);
+
+  useEffect(() => {
+    // Constant pattern to test against
+    const macOSPatterns = /Mac|Macintosh|MacIntel|Mac OS X/;
+
+    // Initial check using userAgent
+    const userAgentCheck = macOSPatterns.test(window.navigator.userAgent);
+
+    setIsMac(userAgentCheck);
+  }, []);
+
+  return isMac;
+}
```

Comment:
The regular expression doesn't need to be created in the hook. The hook is called `useOSDetection`, but what's it's doing is return whether or not it's macOS.

As mentioned in the issue I opened, I'm not sure it if really needs to be a React hook. It could be a regular function.

```suggestion
// Constant pattern to test against
const MACOS_PATTERNS = /Mac|Macintosh|MacIntel|Mac OS X/;
    
function useIsMac(): boolean {
  const [isMac, setIsMac] = useState<boolean>(false);

  useEffect(() => {
    // Initial check using userAgent
    const userAgentCheck = macOSPatterns.test(window.navigator.userAgent);

    setIsMac(userAgentCheck);
  }, []);

  return isMac;
}
```

Reply:
Thank you, @nickytonline. I initially had the code as a function, but I ran into a scenario where the function needed to be called within a useEffect in the component, and the component was responsible for handling the state. To address this, I decided to refactor it into a custom hook, which abstracts away the complexity and only returns the value that the component requires. This way, the code becomes more modular and reusable.

Do you have any thoughts on this?

_PS: Looking at your suggestion above and also implementing styling fixes/updates._